ksql-api/src/main/java/io/confluent/ksql/api/server/protocol/InsertsStreamArgs.java

```diff
@@ -16,15 +16,13 @@
 package io.confluent.ksql.api.server.protocol;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.errorprone.annotations.Immutable;
 import io.vertx.core.json.JsonObject;
 import java.util.Map;
 import java.util.Objects;
 
 /**
  * Represents the arguments to an insert stream request
  */
-@Immutable
```

Comment:
It would be preferable to have all the protocol model classes immutable.

All you need to do is change the `JsonObject` to `ImmutableMap`.  This is actually preferable, as exposing `properties` is currently exposing mutable internal state, which ain't pretty. 

```java
 public final ImmutableMap<String, Object> properties;

...   

this.properties = properties == null 
    ? ImmutableMap.of()
     : ImmutableMap.copyOf(properties);

```

Reply:
Exposing an ImmutableMap doesn't work as it complains that the value (Object) in the Immutable map is not itself Immutable.

This all starts to get complicated. I also notice that the current KsqlRequest class which does something similar (has properties) is not marked as Immutable:

https://github.com/confluentinc/ksql/blob/master/ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/KsqlRequest.java

Reply:
Using ImmutableMap to take a copy of a map passed into an object is not going to give you the encapsulation you require unless the keys and values in the map are themselves immutable.

I notice quite a few places in the code we're talking "immutable" copies of Map<String, ?> or Map<String, Object> so this pattern current seems broken anyway.

Reply:
Yep, fair enough.